crates/wasmtime/src/runtime/component/instance.rs

```diff
@@ -299,9 +305,48 @@ impl Instance {
         store: &StoreOpaque,
         instance: Option<&ComponentExportIndex>,
         name: &str,
-    ) -> Option<ComponentExportIndex> {
+    ) -> Option<(ComponentItem, ComponentExportIndex)> {
         let data = store[self.0].as_ref().unwrap();
         let index = data.component.lookup_export_index(instance, name)?;
+        let ty = match data.component.env_component().export_items[index] {
+            Export::Instance { ty, .. } => TypeDef::ComponentInstance(ty),
+            Export::LiftedFunction { ty, .. } => TypeDef::ComponentFunc(ty),
+            Export::ModuleStatic { ty, .. } | Export::ModuleImport { ty, .. } => {
+                TypeDef::Module(ty)
+            }
+            Export::Type(ty) => ty,
+        };
+        let item = ComponentItem::from(&store.engine(), &ty, &data.ty());
+        Some((
+            item,
+            ComponentExportIndex {
+                id: data.component_id(),
+                index,
+            },
+        ))
```

Comment:
Since this is a copy of what's in `component.rs`, could there be a crate-private helper which takes the `InstanceType` as an argument and both this and the public component version call into that?

Reply:
Good call, I'll make a `pub(crate) ComponentItem::from_export` that takes a wasmtime-environ::Export which should reduce the duplication to just a line or two.

Reply:
[2b13acd](https://github.com/bytecodealliance/wasmtime/pull/10597/commits/2b13acdf960f8106a20ca96f4928f7ffd9af9973)